client/src/cbltest/api/couchbaseserver.py

```diff
@@ -593,3 +667,386 @@ def stop_xcdr(self, target: "CouchbaseServer", bucket_name: str) -> None:
                         f"http://{self.__hostname}:8091/controller/cancelXDCR/{encoded}",
                     )
                     resp.raise_for_status()
+
+    def add_node(
+        self,
+        node_to_add: "CouchbaseServer",
+        services: list[str] | None = None,
+    ) -> None:
+        """
+        Adds a node to the cluster.
+
+        :param node_to_add: The CouchbaseServer instance representing the node to add
+        :param services: List of services to enable on the node (e.g. ["kv", "index", "n1ql"])
+                        Defaults to ["kv", "index", "n1ql"] (data, index, query)
+        """
+        if services is None:
+            services = ["kv", "index", "n1ql"]
+
+        with self.__tracer.start_as_current_span(
+            "add_node",
+            attributes={
+                "cbl.node.hostname": node_to_add.__hostname,
+                "cbl.node.services": ",".join(services),
+            },
+        ) as span:
+            with requests.Session() as session:
```

Comment:
These are heavyweight and should not be created over and over again.  I think probably you should make this a member of the class given how many methods use it.